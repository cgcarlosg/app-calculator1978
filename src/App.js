import './App.css';

function App() {
  return (
    <div className="App">
    <h1>hi hello how are you?</h1>
    </div>
  );
}

export default App;
