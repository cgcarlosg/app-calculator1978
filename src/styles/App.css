* {
  box-sizing: border-box;
}

body {
  padding: 0;
  margin: 0;
  height: 100vh;
  background-color: #1c1d22;
}
