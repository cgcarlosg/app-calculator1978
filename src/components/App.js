import './App.css';
import React, { useState } from 'react';
import Display from './Display';
import calculate from '../logic/calculate';
import ButtonPanel from './ButtonPanel';

const App = () => {

  const [calculation, setCalculation] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = buttonName => {
    const result = calculate(calculation, buttonName);
    setCalculation({ ...result });
  };

  const result = calculation;

  return (
    <>
      <div className="App">
        <header className="App-header">
       <Display result={result} />
          <ButtonPanel clickHandler={handleClick} />
        </header>
      </div>
    </>
  );
}

export default App;
