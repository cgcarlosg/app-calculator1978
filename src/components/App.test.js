import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('Calculator snapshot', () => {
  it('Renders the Calculator page', () => {
    const calculator = renderer
      .create(<App />)
      .toJSON();
    expect(calculator).toMatchSnapshot();
  });
});

describe('Calculator methods', () => {
  beforeEach(() => {
    render(<App />);
  });

  it('Adds two numbers', async () => {
    await userEvent.click(screen.getByText('2'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('+'));
    await userEvent.click(screen.getByText('3'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('='));
    const display = await screen.findByText('50');
    expect(display).toBeInTheDocument();
  });

  it('Substract two numbers', async () => {
    await userEvent.click(screen.getByText('6'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('−'));
    await userEvent.click(screen.getByText('3'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('='));
    const display = await screen.findByText('30');
    expect(display).toBeInTheDocument();
  });

  it('Divide two numbers', async () => {
    await userEvent.click(screen.getByText('7'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('÷'));
    await userEvent.click(screen.getByText('2'));
    await userEvent.click(screen.getByText('='));
    const display = await screen.findByText('35');
    expect(display).toBeInTheDocument();
  });

  it('Divide a number by 0', async () => {
    await userEvent.click(screen.getByText('4'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('÷'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('='));
    const display = await screen.findByText('Math Error');
    expect(display).toBeInTheDocument();
  });

  it('Multiply two numbers', async () => {
    await userEvent.click(screen.getByText('4'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('×'));
    await userEvent.click(screen.getByText('2'));
    await userEvent.click(screen.getByText('='));
    const display = await screen.findByText('80');
    expect(display).toBeInTheDocument();
  });

  it('Convert to negative', async () => {
    await userEvent.click(screen.getByText('9'));
    await userEvent.click(screen.getByText('±'));
    const display = await screen.findByText('-9');
    expect(display).toBeInTheDocument();
  });

  it('Convert to negative', async () => {
    await userEvent.click(screen.getByText('8'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('±'));
    await screen.findByText('-80');
    await userEvent.click(screen.getByText('±'));
    const display = await screen.findByText('80');
    expect(display).toBeInTheDocument();
  });

  it('Change to oposite operation (-)', async () => {
    await userEvent.click(screen.getByText('5'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('−'));
    await userEvent.click(screen.getByText('±'));
    const display = await screen.findByText('+');
    expect(display).toBeInTheDocument();
  });

  it('Change to oposite operation (+)', async () => {
    await userEvent.click(screen.getByText('5'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('+'));
    await userEvent.click(screen.getByText('±'));
    const display = await screen.findByText('−');
    expect(display).toBeInTheDocument();
  });

  it('Convert to percentage', async () => {
    await userEvent.click(screen.getByText('6'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('%'));
    const display = await screen.findByText('0.6');
    expect(display).toBeInTheDocument();
  });

  it('Clear all', async () => {
    await userEvent.click(screen.getByText('4'));
    await userEvent.click(screen.getByText('0'));
    await userEvent.click(screen.getByText('AC'));
    expect(screen.queryByText('40')).toBeNull();
  });
});
