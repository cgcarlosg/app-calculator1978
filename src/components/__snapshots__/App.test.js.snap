// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator snapshot Renders the Calculator page 1`] = `
<div
  className="App"
>
  <div
    className="App-calculator"
  >
    <div
      className="Display"
    >
      
    
    0
    </div>
    <div
      className="btnCont"
    >
      <button
        className="btn btn0"
        name="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn btn1"
        name="±"
        onClick={[Function]}
        type="button"
      >
        ±
      </button>
      <button
        className="btn btn2"
        name="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn btn3"
        name="÷"
        onClick={[Function]}
        type="button"
      >
        ÷
      </button>
    </div>
    <div
      className="btnCont"
    >
      <button
        className="btn btn0"
        name="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn btn1"
        name="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn btn2"
        name="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn btn3"
        name="×"
        onClick={[Function]}
        type="button"
      >
        ×
      </button>
    </div>
    <div
      className="btnCont"
    >
      <button
        className="btn btn0"
        name="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn btn1"
        name="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn btn2"
        name="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn btn3"
        name="−"
        onClick={[Function]}
        type="button"
      >
        −
      </button>
    </div>
    <div
      className="btnCont"
    >
      <button
        className="btn btn0"
        name="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn btn1"
        name="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn btn2"
        name="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn btn3"
        name="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="btnCont"
    >
      <button
        className="btn btn0"
        name="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn btn1"
        name="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn btn2"
        name="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
